torch/csrc/jit/constants.cpp

```diff
@@ -20,8 +20,10 @@ Value* insertConstant(
       n->destroy();
       return g.insertNode(g.createUndefined())->output();
     }
-    if (ref.is_variable()) {
-      ref = autograd::Variable(ref).data();
+    if (!ref.is_variable()) {
```

Comment:
I'd prefer we just `AT_CHECK(ref.is_variable())` and fix the places that pass something else, but this is ok for now if you put the rest in a follow up patch.

Reply:
Added comment in code:
```
// TODO: fix all cases where we are not passing in a variable,
// and then change this to an AT_ASSERT
```